protostar/commands/deploy/network_config.py

```diff
@@ -0,0 +1,38 @@
+from starkware.starknet.cli.starknet_cli import NETWORKS
+
+from protostar.protostar_exception import ProtostarException
+from protostar.utils.config.project import Project
+
+
+class InvalidNetworkConfigException(ProtostarException):
+    pass
+
+
+class NetworkConfig:
+    @classmethod
+    def from_config_file(cls, network_name: str, project: Project) -> "NetworkConfig":
```

Comment:
I don't like we access config in different ways in each module. Can we use `ArgumentValueFromConfigProvider` to access config data?

Reply:
No. Network configuration isn't part of the CLI interface. Also, there might be multiple network configurations.

Maybe I should move network configuration and make it part of deploy command interface.

```toml
[protostar.deploy]
# ...

[protostar.deploy.testnet]
# ...
```
`protostar deploy --profile testnet`

Reply:
Yup i think it should be part of deplot command interface, it would be consistent with other commands

Reply:
If current interface to access the config is not flexible enough, we need to change it. I don't want a different interface for each command, because it's slightly different. If you need my help with that let me know.

Reply:
Added configuration profiles in the separate PR: https://github.com/software-mansion/protostar/pull/291. (Foundry also supports configuration profiles, but in slightly different way).